components/organisms/Chat/ChatMessages/CodeBlock.tsx

```diff
@@ -0,0 +1,93 @@
+import { Button } from "@/components/ui/Button";
+import { useContext, useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { ChatContext } from "./ChatContext";
+import { SalesforceQueryResultWithError } from "@/shared/types/salesforceTypes";
+
+type CodeBlockProps = {
+  code: string;
+  setQueryResult: (arg: SalesforceQueryResultWithError) => void;
+  queryResult: SalesforceQueryResultWithError;
+};
+
+export const CodeBlock = ({
+  code,
+  setQueryResult,
+  queryResult,
+}: CodeBlockProps) => {
+  const [isLoadingQuery, setIsLoadingQuery] = useState(false);
+  const { append, isLoading } = useContext(ChatContext);
+
+  const runQuery = async () => {
+    setIsLoadingQuery(true);
+    const salesforceId = localStorage.getItem("salesforceId");
```

Comment:
This smells - consider using some sort of global state library (like recoil)_ or just a context provider that tracks global salesforce Id